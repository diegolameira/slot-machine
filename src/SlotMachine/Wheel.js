import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

class Wheel extends PureComponent {
  static propTypes = {
    className: PropTypes.string,
    items: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired,
  };

  render() {
    const {
      props: { className, items },
    } = this;
    return (
      <div className={className}>
        {items.map(item => (
          <WheelItem key={item} img={item} />
        ))}
      </div>
    );
  }
}

export const WheelItem = styled.img.attrs(({ img }) => ({
  src: `/imgs/${img}.jpg`,
}))`
  position: absolute;
  background: green;
  width: 80%;
  height: 80%;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
`;

export default styled(Wheel)`
  position: relative;
  overflow: hidden;
  width: 100%;
  height: 0;
  padding-bottom: 33%;
`;
