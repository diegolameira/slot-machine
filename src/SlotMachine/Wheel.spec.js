import React from 'react';
import TestRenderer from 'react-test-renderer';

import Wheel from './Wheel';

describe('Wheel', () => {
  let component = TestRenderer.create(<Wheel items={[]} />);

  it('should render properly', () => {
    expect(component).toMatchSnapshot();
    component.update(<Wheel items={['1', '2']} />);
    expect(component).toMatchSnapshot();
  });
});
