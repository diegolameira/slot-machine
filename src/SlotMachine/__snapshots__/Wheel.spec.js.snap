// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wheel should render properly 1`] = `
<div
  className="sc-bwzfXH eJWyDC"
/>
`;

exports[`Wheel should render properly 2`] = `
<div
  className="sc-bwzfXH eJWyDC"
>
  <img
    className="sc-bdVaJa eqvFIQ"
    src="/imgs/1.jpg"
  />
  <img
    className="sc-bdVaJa eqvFIQ"
    src="/imgs/2.jpg"
  />
</div>
`;
